docs/contributing/coding-style.md

```diff
@@ -297,6 +297,51 @@ All errors should define in [internal/errors package](https://github.com/vdaas/v
 
 Please use [internal/errgroup](https://github.com/vdaas/vald/blob/master/internal/errgroup) for synchronized error handling on multi-goroutine processing.
 
+### Error checking
+
+All functions return `error` if the function can fail. It is very important to ensure the error checking is performed.
+To reduce human mistake that missing the error checking, please check the error using the following style:
+
+```golang
+// good
+if err := fn(); err != nil {
+    // handle error
+}
+```
+
+Instead of this style.
+
+```golang
+// bad
+err := fn()
+if err != nil {
+    // handle error
+}
+```
+
+If you need the value outside the if statement, please use the following style:
+
+```golang
+// good
+conn, err := net.Dial("tcp", "localhost:80")
+if err != nil {
+    // handle error
+}
+
+// use the conn
+```
+
+Instead of this style.
+
+```golang
+// bad
+if conn, err := net.Dial("tcp", "localhost:80");  err != nil {
```

Comment:
I think when logic in the `else` is log, this guideline is correct.
but when logic in the `else` is short, I think the following example is correct.

```
if conn, err := net.Dial(); err != nil {

} else {
    return conn.Close()
}
```

Reply:
in your case I think maybe this is better

if conn, err := net.Dial(); err == nil {
    return conn.Close()
}

// next

Reply:
@kevindiu 
your suggestion seems like having no error handling 🤔

Reply:
@vankichi error handling is the `next` block.

```go
if conn, err := net.Dial(); err == nil {
    return conn.Close()
}

// error handling
```

Reply:
ok
it seems good

Reply:
I'm not sure why we should use this.
IMO, I just use only @hlts2 's suggestion.

Reply:
if the `if block` become huge, it makes code difficult to read. In @hlts2 suggestion,
```go
if conn, err := net.Dial(); err != nil {
    // many
    // lines
    // of
    // logic
} else {
    return conn.Close()
}
```

IMO, it is harder to read than my suggestion.

Reply:
umm...
I could not understand the benefit.
If it's hard to read, we should write as below.
```go
conn, err := net.Dial()
if err != nil {
  // error handling
}
// next
```

Reply:
ah, is it a bad pattern?

Reply:
I think it is a good pattern, like what I wrote in the document. 
```
322 If you need the value outside the if statement, please use the following style:
323
324 ```go
325 // good
326 conn, err := net.Dial("tcp", "localhost:80")
327 if err != nil {
328     // handle error
329 }
330
331 // use the conn
332 ```
```

Seems the origin question from @hlts2 is not a good suggestions, what do you think?

Reply:
i misunderstood, sorry.